composer.json

```diff
@@ -29,6 +29,7 @@
         "sort-packages": true
     },
     "autoload": {
+        "files": [ "src/polyfill.php" ],
```

Comment:
Just wondering: could the polyfill be loaded via a `require_once` at the point where it is actually needed (Json report), instead of globally loaded ?

This would prevent potential issues with other dev tools which may use an `interface_exists()` check on `JsonSerializable` which would now suddenly start passing for them due to the file being automatically loaded (even when this tool is not being run).

Reply:
@jrfn, no, the current codebase is too tightly integrated right now, this class is required for every run, not just `--json`.

so this is something to consider in the future. and I see there's some psr4 pr, that addresses some of the problems.

Reply:
While I understand where you are coming from, that is easily mitigated as the `require_once` could just be added to the application bootstrap file `parallel-lint` in that case, wrapped in an `interface_exists()`.

I still see no reason to always autoload a file which will be rarely needed by this application and may adversely impact other applications when loaded.

Reply:
@jrfnl I don't understand what problem are you trying the solve here.

the executed code is exactly that:

```php
 if (!interface_exists('JsonSerializable', false)) {
    interface JsonSerializable
```

and as I said earlier, the `JsonSerializable` is required for each run of the application regardless if `--json` is used or not:
- https://github.com/php-parallel-lint/PHP-Parallel-Lint/pull/51#discussion_r593939469

Reply:
3f2cc26

Reply:
Thank you ❤️